### PR TITLE
Compatibility with accountsync 5.56

### PR DESCRIPTION
**Pull request by mattwire**
This will require minimum version of accountsync = 5.56. Current version of quickbooks extension (invoice jobs) won't work with new accountsync extension without this change.

### CODE REVIEW

**Comment by agileware-fj** [comment]
Thanks, @mattwire

So, just to confirm - the AccountSync v3 API is going away and has to be replaced with v4 immediately with no deprecation window?

File: CRM/Civiquickbooks/Invoice.php [hunk 1]
```diff
@@ -933,61 +935,38 @@ protected function generateTaxDetails($line_items) {
    * @throws \CiviCRM_API3_Exception
    */
   protected function findPushContributions($params, $limit) {
-    $criteria = [
-      'accounts_needs_update' => 1,
-      'plugin' => $this->plugin,
-      'connector_id' => 0,
-      'accounts_status_id' => ['NOT IN', 3],
```
Comment (mattwire) [inline CRM/Civiquickbooks/Invoice.php:940]: @agileware-fj It was this line that was breaking things I think because 3 is not an array

**Comment by mattwire** [comment]
> Thanks, @mattwire
> 
> So, just to confirm - the AccountSync v3 API is going away and has to be replaced with v4 immediately with no deprecation window?

API3 should still be there for a while. But I took the opportunity to convert at the same time.

Anyway best to hold off on merging this until there is a new release of accountsync.

**Comment by agileware-fj** [comment]
@mattwire Great, the idea of a required to suddenly flick over was slightly worrying.

I'm inclined to have a deprecation window with the APIv3 implementation - with an `accounts_status_id` fix - remaining as a fallback when `Civi\Api4\AccountInvoice` does not exist.

**Comment by axelady** [comment]
I just updated to Account Sync 1.5 and encounter this issue.  First commit did not fix.  With both commits, it now pushes the invoice, but fails with the error 
`Failed to push Contribution: 7141 (AccountInvoice: 3891) Invoice:  with error: '0' is not a valid option for field accounts_status_id.",`
It then keeps creating a new invoice on the quickbooks side with every run.  It looks like I may need to roll back but will poke around a bit more.

**Comment by axelady** [comment]
I've gotten this to work by adding 
``$record['accounts_status_id'] = 1;``
in line 93 just before the call  to AccountSync
`CRM_Accountsync_Hook::accountPushAlterMapped('invoice', $record, $proceed, $accountsInvoice);
`
I don't think this is the "correct" fix because if the call returns $proceed as TRUE then the record will be saved with accounts_status_id = 1 and will not be tried again on error.  

It seems like the issue may be here https://github.com/eileenmcnaughton/nz.co.fuzion.accountsync/blob/87e4a98fea235172fca747dfa97d7feb38314f0d/CRM/Accountsync/BAO/AccountInvoice.php#L297 where accountsync is expecting Xero to return one of these values but Quickbooks doesn't do that and the value remains 0?

**Pull request merged**